modules/cudaoptflow/include/opencv2/cudaoptflow.hpp

```diff
@@ -457,6 +457,129 @@ class CV_EXPORTS_W NvidiaOpticalFlow_1_0 : public NvidiaHWOpticalFlow
         Stream& outputStream = Stream::Null());
 };
 
+/** @brief Class for computing the optical flow vectors between two images using NVIDIA Optical Flow hardware and Optical Flow SDK 2.0.
+@note
+- A sample application demonstrating the use of NVIDIA Optical Flow can be found at
+opencv_contrib_source_code/modules/cudaoptflow/samples/nvidia_optical_flow.cpp
+- An example application comparing accuracy and performance of NVIDIA Optical Flow with other optical flow algorithms in OpenCV can be found at
+opencv_contrib_source_code/modules/cudaoptflow/samples/optical_flow.cpp
+*/
+
+class CV_EXPORTS NvidiaOpticalFlow_2_0 : public NvidiaHWOpticalFlow
+{
+public:
+
+    /**
+    * \struct NVIDIA_OF_ROI_RECT
+    * Specifies the co-ordinates of the Region Of Interest (ROI)
+    * ROI rects should satisfy below requirements:
+    *   1. NV_OF_ROI_RECT::start_x should align to (32 * NV_OF_INIT_PARAMS::outGridSize)
+    *   2. NV_OF_ROI_RECT::width should align to (32 * NV_OF_INIT_PARAMS::outGridSize)
+    *   3. NV_OF_ROI_RECT::start_y should align to (8 * max(NV_OF_INIT_PARAMS::outGridSize, 2))
+    *   4. NV_OF_ROI_RECT::height should align to (8 * NV_OF_INIT_PARAMS::outGridSize)
+    *   5. NV_OF_ROI_RECT::width >= 32 && NV_OF_ROI_RECT::height >= 16; maximum size 8192x8192
+    *   6. Whole ROI region should be inside of the image
+    * Optical flow vectors out side of ROI are invalid and should not be used.
+    */
+    struct NVIDIA_OF_ROI_RECT
```

Comment:
It make sense to reuse `cv::Rect` structure in public API and convert/check its contents in implementation.

Reply:
Sure. Addressed